### PR TITLE
Minor clean code for listener build

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
This is kind of cleanup, without changing the logic.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 1]
```diff
@@ -719,19 +719,19 @@ allChainsLabel:
 
 			// TODO(yxue) avoid bypassing authN using TCP
 			// Build filter chain options for listener configured with protocol sniffing
-			fcm := listener.FilterChainMatch{}
 			if chain.FilterChainMatch != nil {
-				fcm = *chain.FilterChainMatch
+				filterChainMatch = &(*chain.FilterChainMatch)
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/listener.go]: why &*?

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 2]
```diff
@@ -1058,17 +1058,17 @@ func (configgen *ConfigGeneratorImpl) buildHTTPProxy(node *model.Proxy,
...
-		httpProxyPort = int32(features.DefaultPortHTTPProxy)
-	}
-	// enable HTTP PROXY port if necessary; this will add an RDS route for this port
 	if httpProxyPort == 0 {
-		return nil
+		// make sure http proxy is enabled for 'none' interception.
+		if noneMode {
+			httpProxyPort = int32(features.DefaultPortHTTPProxy)
+		} else {
+			return nil
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/listener.go]: you are changing logic here. it was not in else before

File: pilot/pkg/networking/core/v1alpha3/listener.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: it is actually a shallow copy as previous

File: pilot/pkg/networking/core/v1alpha3/listener.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: nit changing, they both return when the `httpProxyPort == 0`

File: pilot/pkg/networking/core/v1alpha3/listener.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: It is a little tricky, i will add some comments

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: So embarrassed,  it seems this does not do a shallow copy, it still get the `chain.FilterChainMatch` address. I will fix it

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: Need a temporary variable

**Comment by hzxuzhonghu** [comment]
@howardjohn  all fixed, ptal

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 3]
```diff
@@ -1644,21 +1633,6 @@ func (configgen *ConfigGeneratorImpl) onVirtualOutboundListener(
...
 func buildSidecarInboundMgmtListeners(node *model.Proxy, push *model.PushContext, managementPorts model.PortList, managementIP string) []*xdsapi.Listener {
 	listeners := make([]*xdsapi.Listener, 0, len(managementPorts))
-
-	if managementIP == "" {
-		managementIP = "127.0.0.1"
-		addr := net.ParseIP(node.IPAddresses[0])
-		if addr != nil && addr.To4() == nil {
-			managementIP = "::1"
-		}
-	}
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/listener.go]: why this change?

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 4]
```diff
@@ -1644,21 +1633,6 @@ func (configgen *ConfigGeneratorImpl) onVirtualOutboundListener(
...
-
-	// NOTE: We should not generate inbound listeners when the proxy does not have any IPtables traffic capture
-	// as it would interfere with the workloads listening on the same port
-	if node.GetInterceptionMode() == model.InterceptionNone {
-		return nil
-	}
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/listener.go]: why change?

**Comment by hzxuzhonghu** [comment]
Sure, let me clarify for some critical part. And i have not changed any old behavior.

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 5]
```diff
@@ -464,13 +463,11 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListeners(
 			// Traffic sent to our service VIP is redirected by remote
 			// services' kubeproxy to our specific endpoint IP.
 			listenerOpts := buildListenerOpts{
-				push:           push,
-				proxy:          node,
-				proxyInstances: node.ServiceInstances,
-				proxyLabels:    node.WorkloadLabels,
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/listener.go]: The above two fields are no longer used. They are only meaninglessly set.

File: pilot/pkg/networking/core/v1alpha3/listener.go (hunk 3 above)
Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: managementIP  can not be "",  which is guaranteed here
```
	for _, ip := range node.IPAddresses {
		managementPorts := push.ManagementPorts(ip)
		management := buildSidecarInboundMgmtListeners(node, push, managementPorts, ip)
		mgmtListeners = append(mgmtListeners, management...)
	}
```

File: pilot/pkg/networking/core/v1alpha3/listener.go (hunk 4 above)
Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: This mode must not be None here which is guaranteed from the caller
```
func (builder *ListenerBuilder) buildManagementListeners(_ *ConfigGeneratorImpl,
	node *model.Proxy, push *model.PushContext) *ListenerBuilder {

	noneMode := node.GetInterceptionMode() == model.InterceptionNone

	// Do not generate any management port listeners if the user has specified a SidecarScope object
	// with ingress listeners. Specifying the ingress listener implies that the user wants
	// to only have those specific listeners and nothing else, in the inbound path.
	if node.SidecarScope.HasCustomIngressListeners || noneMode {
		return builder
	}
	...
	for _, ip := range node.IPAddresses {
                 ...
		management := buildSidecarInboundMgmtListeners(node, push, managementPorts, ip)
	}
```

File: pilot/pkg/networking/core/v1alpha3/listener_builder.go [hunk 6]
```diff
@@ -203,13 +203,10 @@ func (builder *ListenerBuilder) buildSidecarOutboundListeners(configgen *ConfigG
...
 func (builder *ListenerBuilder) buildManagementListeners(_ *ConfigGeneratorImpl,
 	node *model.Proxy, push *model.PushContext) *ListenerBuilder {
-
-	noneMode := node.GetInterceptionMode() == model.InterceptionNone
-
 	// Do not generate any management port listeners if the user has specified a SidecarScope object
 	// with ingress listeners. Specifying the ingress listener implies that the user wants
 	// to only have those specific listeners and nothing else, in the inbound path.
-	if node.SidecarScope.HasCustomIngressListeners || noneMode {
+	if node.SidecarScope.HasCustomIngressListeners || node.GetInterceptionMode() == model.InterceptionNone {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/listener_builder.go]: @howardjohn already checked here

**Comment by hzxuzhonghu** [comment]
@howardjohn Is that ok now？
